bazel/cas/client.go

```diff
@@ -45,23 +46,18 @@ func IsNotFoundError(err error) bool {
 // Read data as bytes from a CAS. Takes a Resolver for addressing and a bazel Digest to read.
 // Returns bytes read or an error. If the requested resource was not found,
 // returns a NotFoundError
-// If retries > 0, does simple retry attempts when encountering errors
-func ByteStreamRead(r dialer.Resolver, digest *remoteexecution.Digest, retries int) (bytes []byte, err error) {
+func ByteStreamRead(r dialer.Resolver, digest *remoteexecution.Digest, b backoff.BackOff) (bytes []byte, err error) {
 	// skip request processing for empty sha
 	if digest == nil || bazel.IsEmptyDigest(digest) {
 		return nil, nil
 	}
-	if retries < 0 {
-		retries = 0
-	}
-
-	for ; retries >= 0; retries-- {
+	backoff.Retry(func() error {
 		bytes, err = byteStreamRead(r, digest)
-
-		if err == nil || IsNotFoundError(err) {
-			break
+		if IsNotFoundError(err) {
+			return nil
```

Comment:
I guess these returns work because the return values are defined in the function signature, but there should be either 2 values returned or just a naked return, so these read weird. Same for all in this client file.

Reply:
Yeah... little funky because backoff.Retry only takes functions with signature `func() error` so we can't return 2 vals here